src/lsp_client.js

```diff
@@ -0,0 +1,96 @@
+const language_client = require("vscode-languageclient/node");
+const vscode = require("vscode");
+const fs = require("fs");
+
+/** @type {language_client.LanguageClient?} */
+let client = null;
+
+function start() {
+    let config = vscode.workspace.getConfiguration("graphene.languageServer");
+
+    if (!config.get("enable", false)) {
+        // Language server is disabled.
+        return;
+    }
+
+    if (!config.has("path")) {
+        // Path not available.
+        vscode.window.showWarningMessage("Language server is enabled but the path is not set");
+        return;
+    }
+
+    const path = config.get("path");
+    try {
+        fs.accessSync(path, fs.constants.R_OK | fs.constants.X_OK);
+    }
+    catch {
+        // Bad path, this check is for the user's convenience only.
+        vscode.window.showErrorMessage(`Invalid language server path: '${path}'`);
+        return;
+    }
+
+    /** @type {language_client.ServerOptions} */
+    const server_options = {
+        run: {
+            command: path,
+            transport: language_client.TransportKind.pipe,
```

Comment:
How can we log debug errors in pipe transport? Would std error get correctly reported by vscode?

Reply:
Yep, if you look at the code I just print to stdout and that shows up in the `Output` tab of vscode, under `Graphene`.